great_expectations/rule_based_profiler/domain_builder/categorical_column_domain_builder.py

```diff
@@ -0,0 +1,247 @@
+            max_proportion_unique=max_proportion_unique,
+        )
+
+        self._exclude_columns = exclude_columns
+
+    @property
+    def domain_type(self) -> Union[str, MetricDomainTypes]:
+        return MetricDomainTypes.COLUMN
+
+    def _get_domains(
+        self,
+        variables: Optional[ParameterContainer] = None,
+    ) -> List[Domain]:
+        """Return domains matching the selected limit_mode.
+
+        Args:
+            variables: Optional variables to substitute when evaluating.
+
+        Returns:
+            List of domains that match the desired cardinality.
+        """
+
+        batch_ids: List[str] = self._get_batch_ids(variables=variables)
+        validator: Validator = self.get_validator(variables=variables)
+
+        # Here we use a single get_metric call to get column names to build the
+        # rest of the metrics.
+        table_column_names: List[str] = self._get_table_column_names_from_first_batch(
+            validator=validator,
+            batch_ids=batch_ids,
+        )
+
+        metrics_for_cardinality_check: List[
+            Dict[str, List[MetricConfiguration]]
+        ] = self._generate_metric_configurations_to_check_cardinality(
+            batch_ids=batch_ids, column_names=table_column_names
+        )
+
+        candidate_column_names: List[str] = self._columns_meeting_cardinality_limit(
+            validator=validator,
+            table_column_names=table_column_names,
+            metrics_for_cardinality_check=metrics_for_cardinality_check,
+        )
+
+        column_domains: List[Domain] = build_domains_from_column_names(
+            candidate_column_names
+        )
+
+        return column_domains
+
+    def _get_table_column_names_from_first_batch(
+        self,
+        validator: Validator,
+        batch_ids: List[str],
+    ):
+        """Retrieve table column names from the first loaded batch.
+
+        Args:
+            validator: Validator to use in retrieving columns.
+            batch_ids: Batches used in this profiler rule.
+
+        Returns:
+            List of column names from the first loaded batch.
+        """
+
+        if len(batch_ids) == 0:
+            raise ProfilerConfigurationError(
+                "No batch_ids were available, please check your configuration."
+            )
+
+        first_batch_id: str = batch_ids[0]
+
+        table_column_names: List[str] = validator.get_metric(
+            metric=MetricConfiguration(
+                metric_name="table.columns",
+                metric_domain_kwargs={
+                    "batch_id": first_batch_id,
+                },
+                metric_value_kwargs=None,
+                metric_dependencies=None,
+            )
+        )
+
+        if self._exclude_columns is not None:
+            table_column_names = [
+                colname
+                for colname in table_column_names
+                if colname not in self._exclude_columns
```

Comment:
@anthonyburdi For things like `exclude_columns`, I would prefer it to be turned into a property and accessed as such herein.  At least that is the style I have been following -- and it plays well with the `Builder` parent class (and above).  Thanks.

Reply:
@alexsherstinsky I hear you, but I would like to keep this private unless/until we have a good reason to access it outside of the class. Currently it is created at instantiation time and not modified which is simpler until we have a reason to make it accessible. Please let me know if I'm missing something.

Reply:
@anthonyburdi The problem is that once we start using this and try to serialize it, the output may not catch/show it, unless there is an accessor (this is per contract with `Builder` and its parents).

Reply:
Done!